operator-framework/src/test/java/io/javaoperatorsdk/operator/SubResourceUpdateIT.java

```diff
@@ -1,36 +1,41 @@
 package io.javaoperatorsdk.operator;
 
-import io.javaoperatorsdk.operator.sample.subresource.SubResourceTestCustomResource;
-import io.javaoperatorsdk.operator.sample.subresource.SubResourceTestCustomResourceController;
-import io.javaoperatorsdk.operator.sample.subresource.SubResourceTestCustomResourceSpec;
+import java.util.concurrent.TimeUnit;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
 import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
 import io.fabric8.kubernetes.client.DefaultKubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
-import io.javaoperatorsdk.operator.api.Controller;
+import io.javaoperatorsdk.operator.sample.subresource.SubResourceTestCustomResource;
+import io.javaoperatorsdk.operator.sample.subresource.SubResourceTestCustomResourceController;
+import io.javaoperatorsdk.operator.sample.subresource.SubResourceTestCustomResourceSpec;
 import io.javaoperatorsdk.operator.sample.subresource.SubResourceTestCustomResourceStatus;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInstance;
 
-import java.util.concurrent.TimeUnit;
-
 import static io.javaoperatorsdk.operator.IntegrationTestSupport.TEST_NAMESPACE;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.awaitility.Awaitility.await;
 
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 public class SubResourceUpdateIT {
-
+    
     private IntegrationTestSupport integrationTestSupport = new IntegrationTestSupport();
-
+    
+    @BeforeAll
+    void checkFinalizer() {
+        assertThat(HasMetadata.FINALIZER_NAME_MATCHER.reset(ControllerUtils.getDefaultFinalizerIdentifier(new SubResourceTestCustomResourceController())).matches()).isTrue();
```

Comment:
What about `HasMetada.isFinalizerValid(String finalizerName): boolean` instead of exposing the matcher object to the user? The usage would be quite simpler and the implementation details (Regex here) wouldn't be exposed.

Reply:
Good point.

Reply:
Only problem is that this will require a new kube client release…